### PR TITLE
Feature/new api

### PR DESCRIPTION
**Pull request by danholdaway**
## Description

Move to use a single argument API. And clean up the base class quite a bit. 

PR means Yaml changes, sorry! But I think applications as a key word in the Yaml wasn't a good choice and that diagnostics is better and more consistent with the way the code is discuessed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,30 +23,28 @@ The eva package can be installed using pip:
...
+diagnostics:
+  - diagnostic name: ObsCorrelationScatter
     ...
-  - application name: ObsMapScatter
+  - diagnostic name: ObsMapScatter
     ...
 ```
 
-eva can also be invoked from another Python module that passes a dictionary, rather than a Yaml file that is later translated into a dictionary. This is achieved as follows:
+eva can also be invoked from another Python module that passes it a dictionary, rather than a Yaml file that is later translated into a dictionary. This is achieved as follows:
```
Comment (CoryMartin-NOAA) [inline README.md]: be consistent here, `yaml` and `Yaml` both in this section, which is best? Or is it `YAML`?

File: src/eva/eva_base.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        eva_dict = eva_config
+    else:
+        # Create dictionary from the input file
+        with open(eva_config, 'r') as eva_config_opened:
+            eva_dict = yaml.safe_load(eva_config_opened)
+
+    # Get the list of applications
+    try:
+        diagnostic_configs = eva_dict['diagnostics']
+    except Exception:
```
Comment (CoryMartin-NOAA) [inline src/eva/eva_base.py]: KeyError?

**Review by CoryMartin-NOAA — COMMENTED** [review]
couple of comments

File: src/eva/eva_base.py [hunk 3]
```diff
@@ -133,7 +133,7 @@ def eva(eva_config, eva_logger=None):
     # Get the list of applications
     try:
         diagnostic_configs = eva_dict['diagnostics']
-    except Exception:
+    except Exception KeyError:
```
Comment (CoryMartin-NOAA) [inline src/eva/eva_base.py]: Does this run? I was suggesting `except KeyError:`

File: src/eva/utilities/logger.py [hunk 4]
```diff
@@ -42,8 +43,8 @@ def __init__(self, task_name):
 
     def send_message(self, level, message):
 
-        if self.loggerdict[level]:
-            print(level+" "+self.task_name+": "+message)
+        if level == 'ABORT' or self.loggerdict[level]:
```
Comment (CoryMartin-NOAA) [inline src/eva/utilities/logger.py]: Can we do a level.upper() here?

File: src/eva/eva_base.py (hunk 3 above)
Comment (danholdaway) [reply src/eva/eva_base.py]: No, I made a typo. Will remove

File: src/eva/utilities/logger.py (hunk 4 above)
Comment (danholdaway) [reply src/eva/utilities/logger.py]: Can do

Comment (CoryMartin-NOAA) [inline src/eva/utilities/logger.py]:
```suggestion
        if level.upper() == 'ABORT' or self.loggerdict[level]:
```

**Review by CoryMartin-NOAA — APPROVED** [review]

File: setup.py [hunk 5]
```diff
@@ -36,6 +36,8 @@
         'pycodestyle>=2.8.0',
         'netCDF4>=1.5.7',
         'matplotlib>=3.4.3',
+        'cartopy>=0.20.2',
```
Comment (CoryMartin-NOAA) [inline setup.py]: I think this needs to be downgraded, the current version of cartopy doesn't work with the ubuntu geos library

**Pull request merged**